pkg/agent/controller/networkpolicy/allocator_test.go

```diff
@@ -168,69 +172,81 @@ func TestRelease(t *testing.T) {
 	}
 }
 
-func TestWorker(t *testing.T) {
+func TestIdAllocatorWorker(t *testing.T) {
 	rule := &types.PolicyRule{
 		Direction: v1beta2.DirectionIn,
 		From:      []types.Address{},
 		To:        ofPortsToOFAddresses(sets.NewInt32(1)),
 		Service:   nil,
 	}
 	tests := []struct {
-		name              string
-		args              []uint32
-		minDeleteInterval time.Duration
-		rule              *types.PolicyRule
-		expectedID        uint32
-		expectedErr       error
+		name                    string
+		args                    []uint32
+		testAsyncDeleteInterval time.Duration
+		rule                    *types.PolicyRule
+		expectedID              uint32
+		expectedErr             error
 	}{
 		{
-			"delete-rule-with-async-delete-interval",
+			// testMinAsyncDeleteInterval(100ms) is larger than testAsyncDeleteInterval(50ms),
+			// so rule should take at least 100ms to be deleted.
+			"delete-rule-with-test-min-async-delete-interval",
 			nil,
-			5 * time.Millisecond,
+			50 * time.Millisecond,
 			rule,
 			1,
 			nil,
 		},
 		{
-			"delete-rule-with-flow-poll-interval",
+			// testAsyncDeleteInterval(200ms) is larger than testMinAsyncDeleteInterval(100ms),
+			// so rule should take at least 200ms to be deleted.
+			"delete-rule-with-test-async-delete-interval",
 			nil,
-			1 * time.Millisecond,
+			200 * time.Millisecond,
 			rule,
 			1,
 			nil,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			minAsyncDeleteInterval = tt.minDeleteInterval
-			a := newIDAllocator(testDeleteInterval, tt.args...)
+			minAsyncDeleteInterval = testMinAsyncDeleteInterval
+			testAsyncDeleteInterval = tt.testAsyncDeleteInterval
+			a := newIDAllocator(testAsyncDeleteInterval, tt.args...)
 			actualErr := a.allocateForRule(tt.rule)
 			if actualErr != tt.expectedErr {
 				t.Fatalf("Got error %v, expected %v", actualErr, tt.expectedErr)
 			}
 			stopCh := make(chan struct{})
 			defer close(stopCh)
-			go wait.Until(a.worker, time.Millisecond, stopCh)
+			defer a.deleteQueue.ShutDown()
+			go wait.Until(a.worker, 50*time.Millisecond, stopCh)
 
-			start := time.Now()
 			a.forgetRule(tt.rule.FlowID)
+
+			startTime := time.Time{}
```

Comment:
seems like you should just set `startTime := time.Now()`, just after calling `forgetRule`?

Reply:
I was playing with time variable and changed it there. Forgot to revert it back. Thanks for catching.